private/PROPOSAL/Propagator.cxx

```diff
@@ -801,13 +799,27 @@ Geometry* Propagator::ParseGeometryConifg(const std::string& json_object_str)
 
     Vector3D vec(x, y, z);
 
+    double hierarchy = geometry->GetHierarchy();
+    if (json_object.find("hierarchy") != json_object.end()) {
+        if (json_object["hierarchy"].is_number()) {
+            hierarchy = json_object["hierarchy"].get<int>();
+        } else {
+            log_fatal(
+                "Invalid input for option 'hierarchy'. Expected a number.");
+        }
+    } else {
+        log_debug("The 'hierarchy' option is not set. Use default (0)");
+    }
+
     // --------------------------------------------------------------------- //
     // Check type of geometry and set members
     // --------------------------------------------------------------------- //
-
-    if (PROPOSAL::Sphere* sphere = dynamic_cast<PROPOSAL::Sphere*>(geometry)) {
+    
+    if (geometry->GetName() == "Sphere") {
```

Comment:
This looks like it should be a switch and name an enum?

Reply:
Thank you, I was not aware of the implicit conversion in smart pointers.

Reply:
I think the problem is solved by commit b98a122 .?